### PR TITLE
Turn on the preview version banner

### LINKED ISSUE #1373: Side-by-side deployment of Renku new and old UIs
# Description

For a transition period, we want allow people to try out the new UI while still leaving the existing UI as the default.

# Configuration / Deployment

There are several options to deploy the two UIs side-by-side.

1. Create a new k8s deployment (beta.renkulab.io) and deploy the new UI there

This would work similarly to how we do the personal dev deployments. beta.renkulab.io would be configured to share gitlab with renkulab.io, but it would have parallel infrastructure for everything else.

**Pros** Easy to set up, easy to run the Selenium tests against both UIs
**Cons** Different URLs for the two versions, cannot seamlessly switch (e.g., sessions started in the old UI would only be visible there)
**Assessment: This is probably the best approach if the resources we need are available.**

2. K8s canary deployment for the new UI

The new and old UIs would be available under the same URL, but a cookie would determine which UI the user gets routed to.

https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#canary

**Pros** Can seamlessly switch between the new and old UI
**Cons** Some work necessary to get it set up; Selenium tests could be complicated

**Assessment: This is a bit more complicated than 1, but would probably work.**

3. Have both UIs available in the UI pod, use nginx to route to the appropriate UI

The new and old UIs would be available under the same URL and Nginx would look at a cookie to determine which one to serve.

~~https://www.nginx.com/blog/performing-a-b-testing-nginx-plus/~~
https://stackoverflow.com/questions/27376235/routed-traffic-with-cookies-nginx

Pros: All setup is restricted to the UI component, can seamlessly switch between the new and old UIs
Cons: Some work necessary to get it set up; Selenium tests could be complicated

**Assessment: About the same as 2 in terms.**

# Old UI Banner

We need to add a banner to the old UI to let people know that they can try out the new one.

image

# New UI Landing

We need to add a new transitionary landing page for the Renku UI that explains the big changes

(Design coming)

### PR DESCRIPTION
**Pull request by ciyer**
Fix #1373

# Testing

Connect to the deployment. 
- Click on the "Try out a preview of the new RenkuLab UI" link. You should be redirected to the new UI.
- Close your browser and come back to the URL. You should still see the new UI
- Delete the "other-ui" cookie and refresh. You should be back to the old UI.

# Appearance

image

/deploy

### CODE REVIEW

**Comment by lorenzo-cavazzi** [comment]
Something doesn't work, if I click on the banner the UI crashes with an error on `homeCustomized`
```
TypeError: e.homeCustomized is undefined
```

Screenshot_20210608_080425

More in general, would it make sense to add the banner also on the logged user home?
I generally don't even look at the landing page and click directly on login, I'm wondering if other users do the same

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
Something doesn't work, if I click on the banner the UI crashes with an error on `homeCustomized`
```
TypeError: e.homeCustomized is undefined
```

Screenshot_20210608_080425

More in general, would it make sense to add the banner also on the logged user home?
I generally don't even look at the landing page and click directly on login, I'm wondering if other users do the same

**Comment by ciyer** [comment]
Can you try it with a private browser session and tell me if you have the same crash?

**Comment by ciyer** [comment]
> Can you try it with a private browser session and tell me if you have the same crash?

Don't worry about it, I have found out the source of the problem (caching). I will fix it.

**Comment by ciyer** [comment]
> Something doesn't work, if I click on the banner the UI crashes with an error on `homeCustomized`

I was able to recreate and fix this problem.

> More in general, would it make sense to add the banner also on the logged user home?
> I generally don't even look at the landing page and click directly on login, I'm wondering if other users do the same

I think you are right, but I would prefer to get this released, so that users that are very eager to try out the new UI have a way to do so, and implement your suggestion in another Issue (#1384) / PR.

File: helm-chart/renku-ui/values.yaml [hunk 1]
```diff
@@ -124,7 +124,7 @@ homepage:
   # Projects to highlight on the homepage
   projects: []
 canary:
-  enabled: false
+  enabled: true
   image:
     repository: renku/renku-ui
-    tag: "0.11.12"
+    tag: "0.11.11-n045.h8e3df79"
```
Comment (lorenzo-cavazzi) [inline helm-chart/renku-ui/values.yaml:130]: Is this intended?

**Comment by rokroskar** [comment]
@ciyer could we have a way to get back to the old UI? right now it seems to be a one-way sort of thing?

**Comment by ciyer** [comment]
> @ciyer could we have a way to get back to the old UI? right now it seems to be a one-way sort of thing?

That will come with #1374. It is not possible to address that in this PR because the changes need to go to different branches.

File: helm-chart/renku-ui/values.yaml (hunk 1 above)
Comment (ciyer) [reply helm-chart/renku-ui/values.yaml:130]: That is the current tag of the ui-design-2021 branch. Before we tag the 0.11.13 release, we should tag the ui-design-2021 as `1.0.0-preview1`, and we can use that tag in this slot.

**Review by lorenzo-cavazzi — APPROVED** [review]
:+1:

**Pull request merged**